src/DB/Query.php

```diff
@@ -287,8 +306,14 @@ protected function build_query( bool $get_count = false ): string {
 
 		$pieces = array_merge( $pieces, $this->generate_where_pieces() );
 
+		if ( ! empty( $this->groupby ) ) {
+			$pieces[] = 'GROUP BY ' . implode( ', ', $this->groupby );
+		}
+
 		if ( ! $get_count ) {
-			$pieces[] = "GROUP BY `{$this->table->get_name()}`.`{$this->table->get_primary_column()}`";
+			if ( empty( $this->groupby ) ) {
+				$pieces[] = "GROUP BY `{$this->table->get_name()}`.`{$this->table->get_primary_column()}`";
```

Comment:
I know we discussed this before somewhere (I think when the get_count functionality was added), but it seems it was still left in the code.

All the primary columns in our tables are unique, so grouping by a unique column doesn't change the result. Do we still want to leave this default group by in place?

Reply:
I also don't think that's necessary but just for the sake of not making this PR more complicated, I left it as is. I'm not sure if it'll have any side effects but it's defined in a common class so maybe it's best to change that in another PR so that we can test it appropriately.